### ISSUE TITLE
`unnecessary_to_owned` suggestion cause error when interacting with `println!` macro

### ISSUE THREAD

**Issue by GitPinkRabbit** [rust-lang/rust-clippy#12821]
### Summary

Clippy thinks the left curly bracket `{` in `println!("{c}")` is an `&`.

### Lint Name

unnecessary_to_owned

### Reproducer

I tried this code:

```rust
fn main() {
    let v: Vec<_> = "hello".chars().collect();
    for c in v.iter().cloned() {
        println!("{c}");
    }
}
```

I saw this happen:

```rust
warning: unnecessary use of `cloned`
 --> src\main.rs:3:14
  |
3 |     for c in v.iter().cloned() {
  |              ^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_to_owned
  = note: `#[warn(clippy::unnecessary_to_owned)]` on by default
help: use
  |
3 |     for c in v.iter() {
  |              ~~~~~~~~
help: remove this `&`
  |
4 -         println!("{c}");
4 +         println!("c}");
  |
```

I expected to see this happen:

Do not suggest to remove the `&`.

### Version

```text
rustc 1.75.0 (82e1608df 2023-12-21)
binary: rustc
commit-hash: 82e1608dfa6e0b5569232559e3d385fea5a93112
commit-date: 2023-12-21
host: x86_64-pc-windows-msvc
release: 1.75.0
LLVM version: 17.0.6
```

### Additional Labels

@rustbot label I-suggestion-causes-error
